src/processor/mod.rs

```diff
@@ -157,9 +157,12 @@ impl TransactionContext for EmptyTransactionContext {
 
     fn get_state_entries_by_prefix(
         &self,
+        tip_id: &str,
```

Comment:
I'd prefer if this was `Option<&str>`, and we would fall back to the default of `""` if `None` is provided.

Reply:
not sure i understand: do you want to pass None instead of "" from a caller and then inside the calee have an additional test to convert None to ""?